### PR TITLE
migrate file to jsonl on setup

### PR DESCRIPTION
**Pull request by foxriver76**

### CODE REVIEW

**Comment by Apollon77** [comment]
@foxriver76 I think we also need it for "setup firs"t which is executed on controller 4 install

File: packages/controller/lib/setup.js [hunk 1]
```diff
@@ -600,82 +600,109 @@ async function processCommand(command, args, params, callback) {
...
+                                try {
+                                    // TODO
+                                    await notificationHandler.addMessage(
+                                        'system',
+                                        'memIssues',
+                                        '',
+                                        `system.host.${hostname}`
+                                    );
+                                } catch (e) {
+                                    console.warn(`Could not add OOM notification: ${e.message}`);
```
Comment (Apollon77) [inline packages/controller/lib/setup.js:693]: OOM? :-)

File: packages/controller/lib/setup.js [hunk 2]
```diff
@@ -600,82 +600,109 @@ async function processCommand(command, args, params, callback) {
...
+                                const notificationHandler = new NotificationHandler(notificationSettings);
 
-                                if (!isDeepStrictEqual(config, configOrig)) {
-                                    fs.writeFileSync(configFile, JSON.stringify(config, null, 2));
-                                    console.log('ioBroker configuration updated');
+                                try {
+                                    // TODO
+                                    await notificationHandler.addMessage(
+                                        'system',
+                                        'memIssues',
```
Comment (Apollon77) [inline packages/controller/lib/setup.js:688]: mem issues?

File: packages/controller/lib/setup.js [hunk 3]
```diff
@@ -600,82 +600,109 @@ async function processCommand(command, args, params, callback) {
...
-                            try {
-                                await objects.delObjectAsync('system.adapter.js-controller');
-                            } catch {
-                                // ignore
+                            let migrated = false;
+                            // We migrate file to jsonl
+                            if (config.states.type === 'file') {
+                                config.states.type = 'jsonl';
+                                console.log('States DB type migrated from "file" to "jsonl"');
+                                migrated = true;
```
Comment (Apollon77) [inline packages/controller/lib/setup.js:660]: maybe remember which db was cchanged to add to the notification?

**Review by Apollon77 — CHANGES_REQUESTED** [review]

File: packages/controller/lib/setup.js (hunk 2 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:688]: draft, see todo ;-)

File: packages/controller/lib/setup.js (hunk 1 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:693]: draft lol

File: packages/controller/lib/setup.js (hunk 3 above)
Comment (foxriver76) [reply packages/controller/lib/setup.js:660]: yes, next steps

**Pull request merged**